src/lib/support/ZclString.h

```diff
@@ -0,0 +1,36 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <app/util/af-types.h>
+
+namespace chip {
+
+/**
+ * @brief Create ZCL string from char
+ */
+template <size_t BufferLength>
+uint8_t * MakeZclCharString(uint8_t (&zclString)[BufferLength], char * cString)
+{
+    static_assert(BufferLength <= 254, "Too long string to fit in ZCL_CHAR_STRING type");
```

Comment:
I made a mistake earlier... the buffer we are writing to can effectively be up to 255 bytes: one byte length and 254 bytes data.  The buffer we copy from can't be bigger than 254 bytes for this to make sense.

So this assert should be:
```suggestion
    static_assert(BufferLength <= 255, "Too long string to fit in ZCL_CHAR_STRING type");
```

Reply:
Updated.